crates/utils/re_arrow_util/src/arrow_util.rs

```diff
@@ -248,8 +250,9 @@ pub fn new_list_array_of_empties(child_datatype: &DataType, len: usize) -> ListA
 /// Returns an error if the arrays don't share the exact same datatype.
 pub fn concat_arrays(arrays: &[&dyn Array]) -> arrow::error::Result<ArrayRef> {
     #[allow(clippy::disallowed_methods)] // that's the whole point
-    arrow::compute::concat(arrays)
-    // TODO(#3741): call .shrink_to_fit on the result
+    let mut array = arrow::compute::concat(arrays)?;
+    array.shrink_to_fit(); // VERY IMPORTANT! https://github.com/rerun-io/rerun/issues/7222
+    Ok(array)
```

Comment:
Reminder: everything using a compute kernel in this file requires the same treatment.